File: integration-tests/kafka/src/test/java/io/quarkus/it/kafka/KafkaTestResource.java
package io.quarkus.it.kafka;

import java.util.Collections;
import java.util.Map;

import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
import io.strimzi.StrimziKafkaContainer;

public class KafkaTestResource implements QuarkusTestResourceLifecycleManager {

    private static final StrimziKafkaContainer kafka = new StrimziKafkaContainer();

    public static String getBootstrapServers() {
        return kafka.getBootstrapServers();
    }

    @Override
    public Map<String, String> start() {
        kafka.start();
        return Collections.singletonMap("kafka.bootstrap.servers", kafka.getBootstrapServers());
    }

    @Override
    public void stop() {
        kafka.close();
    }
}

File: integration-tests/resteasy-reactive-kotlin/standard/src/test/kotlin/io/quarkus/it/resteasy/reactive/kotlin/KafkaTestResource.kt
package io.quarkus.it.resteasy.reactive.kotlin

import io.quarkus.test.common.QuarkusTestResourceLifecycleManager
import io.strimzi.test.container.StrimziKafkaCluster

class KafkaTestResource : QuarkusTestResourceLifecycleManager {

    private val kafka: StrimziKafkaCluster =
        StrimziKafkaCluster.StrimziKafkaClusterBuilder().build()

    fun getBootstrapServers(): String? {
        return kafka.getBootstrapServers()
    }

    override fun start(): Map<String, String>? {
        kafka.start()
        return mapOf("kafka.bootstrap.servers" to kafka.getBootstrapServers())
    }

    override fun stop() {
        kafka.close()
    }
}
